trio/_core/_tests/test_instrumentation.py

```diff
@@ -255,7 +255,7 @@ def after_run(self) -> NoReturn:
             raise ValueError("oops")
 
     async def main() -> None:
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError, match="oops"):
```

Comment:
People keep forgetting that it's a regex. Could we make that explicit?
```suggestion
        with pytest.raises(ValueError, match="^oops$"):
```